src/components/NetworkModal/index.tsx

```diff
@@ -67,6 +69,8 @@ export default function NetworkModal(): JSX.Element | null {
             } else {
                 void setChain({ chainId: ChainIdHex[key] })
             }
+
+            sendData({ event: 'network_switch', action: 'network_switch_success', network: ChainId[key] })
```

Comment:
better to listen for the `chainId` variable, e.g. when chain was actually switched
or other variable, maybe useSetChain returns current chain as first arg ?

Reply:
This is not based on connected network, but the one you select in the list of available networks, why key is used
(from the allowedNetworks map 20 lines below)

useSetChain returns only boolean to indicate it has connected
Current chainId from walletState will only be available after render, not immediately after

Reply:
connectedChain (option from useSetChain) is only updated after the callback runs, current chain has no usage here since its about the switch TO chain
so that's why we take here the key from the selected network list

Reply:
you call set chain then immediately send 'success' event
this is not right
should be setChain -> wait chain successfully changed -> send event